tests/test_connector.py

```diff
@@ -557,6 +557,40 @@ async def certificate_error(*args, **kwargs):
     assert isinstance(ctx.value, aiohttp.ClientSSLError)
 
 
+async def test_tcp_connector_server_hostname_default(loop) -> None:
+    conn = aiohttp.TCPConnector(loop=loop)
+
+    conn._loop.create_connection = mock.AsyncMock()
+    conn._loop.create_connection.return_value = mock.Mock(), mock.Mock()
+
+    req = ClientRequest("GET", URL("https://127.0.0.1:443"), loop=loop)
+
+    established_connection = await conn.connect(req, [], ClientTimeout())
```

Comment:
Can probably use `async with` here, rather than needing to close explicitly.

Reply:
Sadly, `.connect` returns a [`Connection`](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/connector.py#L72) which doesn't implement the context management protocol :/

I can use [`contextlib.closing`](https://docs.python.org/3/library/contextlib.html#contextlib.closing) though :+1: